packages/fx-core/src/core/collaborator.ts

```diff
@@ -0,0 +1,495 @@
+  const aadAppTenantId = envInfo.state[BuiltInFeaturePluginNames.appStudio]?.tenantId;
+
+  for (const teamsAppOwner of teamsAppOwners) {
+    const aadOwner = aadOwners.find((owner) => owner.userObjectId === teamsAppOwner.userObjectId);
+
+    collaborators.push({
+      // For guest account, aadOwner.userPrincipalName will be user's email, and is easy to read.
+      userPrincipalName:
+        aadOwner?.userPrincipalName ??
+        teamsAppOwner.userPrincipalName ??
+        teamsAppOwner.userObjectId,
+      userObjectId: teamsAppOwner.userObjectId,
+      isAadOwner: aadOwner ? true : false,
+      aadResourceId: aadOwner ? aadOwner.resourceId : undefined,
+      teamsAppResourceId: teamsAppOwner.resourceId,
+    });
+  }
+
+  if (inputs.platform === Platform.CLI || inputs.platform === Platform.VSCode) {
+    const message = [
+      {
+        content: getStrings().solution.Collaboration.ListingM365Permission,
+        color: Colors.BRIGHT_WHITE,
+      },
+      {
+        content: getStrings().solution.Collaboration.AccountUsedToCheck,
+        color: Colors.BRIGHT_WHITE,
+      },
+      { content: user.userPrincipalName + "\n", color: Colors.BRIGHT_MAGENTA },
+      {
+        content: getStrings().solution.Collaboration.StartingListAllTeamsAppOwners,
+        color: Colors.BRIGHT_WHITE,
+      },
+      { content: `${envInfo.envName}\n`, color: Colors.BRIGHT_MAGENTA },
+      { content: getStrings().solution.Collaboration.TenantId, color: Colors.BRIGHT_WHITE },
+      { content: aadAppTenantId + "\n", color: Colors.BRIGHT_MAGENTA },
+      { content: getStrings().solution.Collaboration.M365TeamsAppId, color: Colors.BRIGHT_WHITE },
+      { content: teamsAppId, color: Colors.BRIGHT_MAGENTA },
+    ];
+
+    if (isAadActivated) {
+      message.push(
+        { content: getStrings().solution.Collaboration.SsoAadAppId, color: Colors.BRIGHT_WHITE },
+        { content: aadAppId, color: Colors.BRIGHT_MAGENTA },
+        { content: `)\n`, color: Colors.BRIGHT_WHITE }
+      );
+    } else {
+      message.push({ content: ")\n", color: Colors.BRIGHT_WHITE });
+    }
+
+    for (const collaborator of collaborators) {
+      message.push(
+        { content: getStrings().solution.Collaboration.TeamsAppOwner, color: Colors.BRIGHT_WHITE },
+        { content: collaborator.userPrincipalName, color: Colors.BRIGHT_MAGENTA },
+        { content: `. `, color: Colors.BRIGHT_WHITE }
+      );
+
+      if (isAadActivated && !collaborator.isAadOwner) {
+        message.push({
+          content: getStrings().solution.Collaboration.NotOwnerOfSsoAadApp,
+          color: Colors.BRIGHT_YELLOW,
+        });
+      }
+
+      message.push({ content: "\n", color: Colors.BRIGHT_WHITE });
+    }
+
+    if (inputs.platform === Platform.CLI) {
+      ctx.userInteraction.showMessage("info", message, false);
+    } else if (inputs.platform === Platform.VSCode) {
+      const hasSPFx = solutionSettings?.activeResourcePlugins?.includes(
```

Comment:
Are there better ways to determine spfx project instead of using AAD? Because in the future, we may have tab app without AAD plugin but it is not SPFX project.

Reply:
Here the name hasSPFx is not corrent, which realy means hasAAD. I just keep the same logic with v2